### ISSUE TITLE
[Feature] Share to Instapaper

### ISSUE THREAD

**Issue by obra** [FreshRSS/FreshRSS#6988]
**Is your feature request related to a problem? Please describe.**

With Omnivore being EOLed and Pocket no longer actually archiving content in some cases, I find myself switching back to Instapaper, one of the classic 'read it later' services.  FreshRSS does not currently have a 'save to instapaper' feature.

**Describe the solution you’d like**

I'd love to see Instapaper added as a sharing service.  While not OSS, Instapaper does have a "free forever" service option. I'm very happy to test out a PR or patch and provide feedback if that would be helpful.

**Describe alternatives you’ve considered**

I've considered whether this missing feature would be enough to get me to switch read-it-later service (again) or to switch feed readers and neither of those is particularly enticing. ;)

**Additional context**

You can find the API docs here: 

https://www.instapaper.com/api/simple

But the basic API is pretty straightforward: 
```

Adding URLs to an Instapaper account
URL: https://www.instapaper.com/api/add

Parameters:

username and password (Or you can pass the username and password via HTTP Basic Auth.)
url
title -- optional, plain text, no HTML, UTF-8. If omitted or empty, Instapaper will crawl the URL to detect a title.
selection -- optional, plain text, no HTML, UTF-8. Will show up as the description under an item in the interface. Some clients use this to describe where it came from, such as the text of the source Twitter post when sending a link from a Twitter client.
redirect=close -- optional. Specifies that, instead of returning the status code, the resulting page should show an HTML "Saved!" notification that attempts to close its own window with Javascript after a short delay. This is useful if you're sending people directly to /api/add URLs from a web application.
jsonp -- optional. See .

**Comment by Frenzie** [comment]
A quick search shows they also have a bookmarklet. https://www.instapaper.com/save
```js
javascript:function iptxt(){var d=document;try{if(!d.body)throw(0);window.location='http://www.instapaper.com/text?u='+encodeURIComponent(d.location.href);}catch(e){alert('Please wait until the page has loaded.');}}iptxt();void(0)
```
As far as implementation goes that's easier than a special API. I suppose it depends on how much you care about having to be logged in while using it.

**Comment by obra** [comment]
> A quick search shows they also have a bookmarklet. https://www.instapaper.com/save
> 
> ```js
> 
> javascript:function iptxt(){var d=document;try{if(!d.body)throw(0);window.location='http://www.instapaper.com/text?u='+encodeURIComponent(d.location.href);}catch(e){alert('Please wait until the page has loaded.');}}iptxt();void(0)
> 
> ```
> 
> As far as implementation goes that's easier than a special API. I suppose it depends on how much you care about having to be logged in while using it.

I believe that bookmarklet is for their "render as text" view. The "save the page" bookmarklet appears to bake a magic identifier into the url, which I'd assumed was a user token, but might just be cache busting? (I personally don't mind it only working if I'm logged in)

**Comment by Frenzie** [comment]
My bad, but it seems there's an "add" bookmarklet as well:
![image](https://github.com/user-attachments/assets/45ae35f0-3504-4c86-93cd-02e231d83f86)

**Comment by obra** [comment]
> My bad, but it seems there's an "add" bookmarklet as well: ![image](https://private-user-images.githubusercontent.com/202757/384725878-45ae35f0-3504-4c86-93cd-02e231d83f86.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEyODgxMDAsIm5iZiI6MTczMTI4NzgwMCwicGF0aCI6Ii8yMDI3NTcvMzg0NzI1ODc4LTQ1YWUzNWYwLTM1MDQtNGM4Ni05M2NkLTAyZTIzMWQ4M2Y4Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExMVQwMTE2NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNTJmMjc3OTNjYjJmM2NkMTg2YmQ3MTk1MTYyZmE2NmNmYjk0NzkwOThhMDEyZTg1NzVmNmRhNTBlZTkwMTNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.YDcB5K_tgAIyStTMKkFNklNGvQOD09M0kKlsb89uiyk)

Yes. That bookmarklet includes a user-specific identifier. (Replaced by ##SECRETUSERID## in the content below):

```javascript:function iprl5(){var d=document,z=d.createElement('scr'+'ipt'),b=d.body,l=d.location;try{if(!b)throw(0);d.title='(Saving...) '+d.title;z.setAttribute('src',l.protocol+'//www.instapaper.com/j/##SECRETUSERID##?a=read-later&u=%27+encodeURIComponent(l.href)+%27&t=%27+(new%20Date().getTime()));b.appendChild(z);}catch(e){alert(%27Please%20wait%20until%20the%20page%20has%20loaded.%27);}}iprl5();void(0)```

**Comment by benjuade** [comment]
The bookmark seems that for some websites is kind of deprecated, but probably only for those that have a paywall (and thus maybe not relevant for FreshRSS).

For both cases (simple API or bookmark), FreshRSS needs to save a secret password/token. So first thing is to check whether it allows secrets to be stored securely.
